specification/security/resource-manager/Microsoft.Security/preview/2022-04-01-preview/query.json

```diff
@@ -0,0 +1,231 @@
+          "200": {
+            "description": "Result of the query operation",
+            "schema": {
+              "$ref": "#/definitions/QueryResponse"
+            }
+          },
+          "default": {
+            "description": "An error occurred while processing the request. See the error.code parameter to identify the specific error.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Basic query": {
+            "$ref": "./examples/BasicQuery.json"
+          },
+          "Complex query": {
+            "$ref": "./examples/ComplexQuery.json"
+          }
+        }
+      }
+    },
+  },
+  "definitions": {
+    "QueryRequest": {
+      "description": "Describes a query to be executed.",
+      "type": "object",
+      "properties": {
+        "scopes": {
+          "description": "The scopes in which the query will be executed - List of subscription IDs for Azure environments, and list of Account IDs for AWS environments.",
+          "type": "array",
+          "items": {
+            "type": "string",
+            "description": "Scope."
+          }
+        },
+        "query": {
+          "description": "The KQL query to execute.",
+          "type": "string"
+        },
+        "options": {
+          "description": "The query evaluation options",
+          "$ref": "#/definitions/QueryRequestOptions"
+        },
+      },
+      "required": [
+        "query"
+      ]
+    },
+	"QueryRequestOptions": {
+      "description": "The options for query evaluation",
+      "properties": {
+        "$skipToken": {
+          "description": "Continuation token for pagination, capturing the next page size and offset, as well as the context of the query.",
+          "type": "string"
+        },
+        "$top": {
+          "description": "The maximum number of rows that the query should return. Overrides the page size when ```$skipToken``` property is present.",
+          "type": "integer",
+          "format": "int32",
+          "minimum": 1,
+          "maximum": 1000
+        },
+        "$skip": {
+          "description": "The number of rows to skip from the beginning of the results. Overrides the next page offset when ```$skipToken``` property is present.",
+          "type": "integer",
+          "format": "int32",
+          "minimum": 0
+        }
+      }
+    },
+	"QueryResponse": {
+      "description": "Query result.",
+      "type": "object",
+      "properties": {
+        "totalRecords": {
+          "description": "Number of total records matching the query.",
+          "type": "integer",
+          "format": "int64"
+        },
+        "count": {
+          "description": "Number of records returned in the current response. In the case of paging, this is the number of records in the current page.",
+          "type": "integer",
+          "format": "int64"
+        },
+        "$skipToken": {
```

Comment:
>$skipToken [](http://example.com/codeflow?start=9&length=10)

SDK folks can comment on whether this is a best practice for paginated POST calls.  The SDK won't follow pages automatically and users will have to write custom loops

Reply:
We have this aligned with other tenant level API in the swagger files - the Microsoft.ResourceGraph resources API.
pagination can be done indeed only by the options properties, therefore custom loops by users, but i believe this is already stable solution